.circleci/config.yml

```
@@ -121,15 +117,13 @@ jobs:
             - source-v1-{{ .Branch }}-
             - source-v1-
       - checkout
-      - run:
-          name: 'Compute and Fetch Merge Commit'
-          command: |
-            ./.circleci/compute_merge_commit.sh
-            ./.circleci/fetch_merge_commit.sh
       - save_cache:
           key: source-v1-{{ .Branch }}-{{ .Revision }}
           paths:
             - .git
+      - run:
+          name: 'Compute and Fetch Merge Commit'
```

nit: Update step name now that you're no longer calling the fetch script here.

```suggestion
          name: 'Compute Merge Commit'
```

---

We're technically still fetching it ;)

---

In that case, the script name has become misleading and should be fixed.

I'm curious about why `compute_merge_commit.sh` is doing any fetching. Why not leave that to `fetch_merge_commit.sh` so that the call to `git pull --ff-only` is not repeated in two scripts?

---

Before `.git` caching was introduced, `compute_merge_commit.sh` would merely determine the SHA of the merge commit meant to be used by the PR (without actually fetching the commits), while `fetch_merge_commit.sh` would only be run by subsequent stages to do the actual fetching of commits. This separation of roles has partially gone away with the introduction of caching (and with this fix).

After our offline convo, I'm satisfied with _what_ each file is doing. My only remaining (minor) concern is that the old filename is confusing. To keep things simple, can we rename `compute_merge_commit.sh` to `initialize_repo.sh` (since it is now doing a whole lot more) and then rename the `Compute and Fetch Merge Commit` step to `Initialize Repository` (since it contains the meat of the job with the same name)?

---

This recommendation LGTM! Done.